contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -1069,26 +1118,30 @@ private void getAndMapCompoundData(String path, List<String> fieldNames, IHDF5Re
           for (int col = 0; col < values[row].length; col++) {
             assert fieldNames != null;
             currentFieldName = fieldNames.get(col);
-            ArrayWriter innerWriter = listWriter.array(currentFieldName);
-            if (values[row][col] instanceof Integer) {
-              innerWriter.scalar().setInt((Integer) values[row][col]);
-            } else if (values[row][col] instanceof Short) {
-              innerWriter.scalar().setInt((Short) values[row][col]);
-            } else if (values[row][col] instanceof Byte) {
-              innerWriter.scalar().setInt((Byte) values[row][col]);
-            } else if (values[row][col] instanceof Long) {
-              innerWriter.scalar().setLong((Long) values[row][col]);
-            } else if (values[row][col] instanceof Float) {
-              innerWriter.scalar().setDouble((Float) values[row][col]);
-            } else if (values[row][col] instanceof Double) {
-              innerWriter.scalar().setDouble((Double) values[row][col]);
-            } else if (values[row][col] instanceof BitSet || values[row][col] instanceof Boolean) {
-              innerWriter.scalar().setBoolean((Boolean) values[row][col]);
-            } else if (values[row][col] instanceof String) {
-              innerWriter.scalar().setString((String) values[row][col]);
-            }
-            if (col == values[row].length) {
-              innerWriter.save();
+            try {
+              ArrayWriter innerWriter = listWriter.array(currentFieldName);
+              if (values[row][col] instanceof Integer) {
+                innerWriter.scalar().setInt((Integer) values[row][col]);
+              } else if (values[row][col] instanceof Short) {
+                innerWriter.scalar().setInt((Short) values[row][col]);
+              } else if (values[row][col] instanceof Byte) {
+                innerWriter.scalar().setInt((Byte) values[row][col]);
+              } else if (values[row][col] instanceof Long) {
+                innerWriter.scalar().setLong((Long) values[row][col]);
+              } else if (values[row][col] instanceof Float) {
+                innerWriter.scalar().setDouble((Float) values[row][col]);
+              } else if (values[row][col] instanceof Double) {
+                innerWriter.scalar().setDouble((Double) values[row][col]);
+              } else if (values[row][col] instanceof BitSet || values[row][col] instanceof Boolean) {
+                innerWriter.scalar().setBoolean((Boolean) values[row][col]);
+              } else if (values[row][col] instanceof String) {
+                innerWriter.scalar().setString((String) values[row][col]);
+              }
```

Comment:
What happens for the "none of the above" case? Should we throw an exception to be safe?

Reply:
I don't want to throw an exception here, but I'll send a warning to the logger instead.